code/modules/admin/smites/imaginary_friend_special.dm

```diff
@@ -10,34 +14,61 @@
  **/
 /datum/smite/custom_imaginary_friend
 	name = "Imaginary Friend (Special)"
-	var/client/friend_candidate_client
+	/// Who are we going to add to your head today?
+	var/list/friend_candidates
+	/// Do we randomise friend appearances or not?
 	var/random_appearance
 
 /datum/smite/custom_imaginary_friend/configure(client/user)
-	friend_candidate_client = tgui_input_list(user, "Pick the player to put in control", "New Imaginary Friend", sort_list(GLOB.clients))
-	if(isnull(friend_candidate_client))
+	var/appearance_choice = tgui_alert(user,
+		"Do you want the imaginary friend(s) to share name and appearance with their currently selected character preferences?",
+		"Imaginary Friend Appearance?",
+		list(CHOICE_PREFS_APPEARANCE, CHOICE_RANDOM_APPEARANCE, "Cancel"))
+	if (appearance_choice != CHOICE_PREFS_APPEARANCE && appearance_choice != CHOICE_RANDOM_APPEARANCE)
```

Comment:
just do `isnull()` and match to `"Cancel"` in a shortcircuited `||` instead. like

`if(isnull(appearance_choice) || appearance_choice = "Cancel"))`

way  more cleaner and likely faster too since we can insta exit without doing two equivalency checks if the admin closes out of the prompt